--- a/plugins/output_viewer/CMakeLists.txt
+++ b/plugins/output_viewer/CMakeLists.txt
@@ -1,8 +1,8 @@
 # SDL detection
-find_package(SDL REQUIRED)
-find_package(SDL_image REQUIRED)
+find_package(SDL2 REQUIRED)
+find_package(SDL2_image REQUIRED)
 
 # Link libraries
 add_library(output_viewer SHARED output_viewer.c)
-target_link_libraries(output_viewer 
-    ${SDL_LIBRARY} 
-    ${SDLIMAGE_LIBRARY})
+target_link_libraries(output_viewer
+    ${SDL2_LIBRARIES}
+    ${SDL2_IMAGE_LIBRARIES})
